content/blog/2023-11-23-announcing-axum-0-7-0.md

```diff
@@ -0,0 +1,184 @@
+that `hyper` and `hyper-util` has. `http-body` now just provides the core APIs
+and high level utilities has been moved to [`http-body-util`]. That includes
+things like `Full`, `Empty`, and `UnsyncBoxBody`, which used to be re-exported by
+`axum`.
+
+For the same reason that `hyper-util` shouldn't be part of `axum`'s public API,
+`http-body-util` shouldn't either.
+
+As a replacement `axum` now provides its own body type found at `axum::body::Body`.
+
+Its API is similar to what `hyper::Body` had:
+
+```rust
+use axum::body::Body;
+use futures::TryStreamExt;
+use http_body_util::BodyExt;
+
+// Create an empty body (Body::default() does the same)
+Body::empty();
+
+// Create bodies from strings or buffers
+Body::from("foobar");
+Body::from(Vec::<u8>::from([1, 3, 3, 7]));
+
+// Wrap another type that implements `http_body::Body`
+Body::new(another_http_body);
+
+// Convert a `Body` into a `Stream` of data frames
+let mut stream = body.into_data_stream();
+while let Some(chunk) = stream.try_next().await? {
+    // ...
+}
+
+// Collect the body into a `Bytes`. Uses `BodyExt::collect`
+// This replaces the previous `hyper::body::to_bytes` function
+let bytes = body.collect().await?.to_bytes();
+```
+
+## Fewer generics
+
+`axum::Router` used to be generic over the request body type. That meant
+applying middleware that changed the request body type would have knock on
+effects throughout your routes:
+
+```rust
+// This would work just fine
+Router::new()
+    .route(
+        "/",
+        get(|body: Request<Body>| async { ... })
+    );
+
+// But adding `tower_http::limit::RequestBodyLimitLayer` would make
+// things no longer compile
+Router::new()
+    .route(
+        "/",
+        get(|body: Request<Body>| async { ... })
+    )
+    .layer(tower_http::limit::RequestBodyLimitLayer::new(1024));
+```
+
+The reason it doesn't work is that `RequestBodyLimitLayer` changes the request
+body type so you have to extract `Request<http_body::Limited<Body>>` instead of
+`Request<Body>`. This was very subtle and the source of some confusion.
+
+However in axum 0.7 everything continues working as before, regardless of which
+middleware you add:
+
+```rust
+Router::new()
+    .route(
+        "/",
+        // You always extract `Request<Body>` no which middleware
+        // you add
+        get(|body: Request<Body>| async { ... })
+    )
+    .layer(tower_http::limit::RequestBodyLimitLayer::new(1024));
+```
+
+There is also a convenient type alias of `Request` that uses `Body`:
+
+```rust
+use axum::extract::Request;
+
+Router::new().route("/", get(|body: Request| async { ... }));
+```
+
+The request body type parameter has been removed from all of `axum`'s types and
+traits including `FromRequest`, `MethodRouter`, `Next`, and more.
+
+## See the changelog for more
+
+I encourage you to read the [changelog] to see all the changes and for tips on
+how to update from 0.6 to 0.7.
+
+Also, please ask questions in [Discord] or [file issues] if you have trouble
```

Comment:
GitHub discussions is missing, IMHO it should be preferred over Discord for questions since the expectation that you'd search for existing discussions first seems higher.